bufferserver/src/main/java/com/datatorrent/bufferserver/server/Server.java

```diff
@@ -807,7 +807,7 @@ private boolean switchToNewBufferOrSuspendRead(final byte[] array, final int off
 
     private boolean switchToNewBuffer(final byte[] array, final int offset, final int size)
     {
-      if (datalist.isMemoryBlockAvailable()) {
+      if (datalist.isMemoryBlockAvailable() || datalist.canAllocateMemoryBlock()) {
```

Comment:
I thought about that but since isMemoryBlockAvailable is computationally less expensive than  areSubscribersBehind and when memory block is available even in the spooling disabled case the publisher can proceed, chose to do this instead.